### PR TITLE
feat: STAKE-803 integrate unstake method from sdk

### PR DESCRIPTION
**Pull request by Matt561**
## **Description**
This PR adds the ability to unstake natively within the MetaMask mobile app.

## **Related issues**

Jira Ticket: [FE] [Unstake flow] Integrate unstake method from sdk - ([link](https://consensyssoftware.atlassian.net/browse/STAKE-803))

## **Manual testing steps**

prerequisite: if you don't have any Ethereum staked, you'll need to stake some in order to unstake.

1. Add `export MM_POOLED_STAKING_UI_ENABLED=true` to `.js.env` file.
2. Click on Ethereum from the asset list page
3. Click on "Unstake"
4. Enter a valid amount to unstake and click "Review"
5. Click "Continue" to initiate the transaction-controller flow.

## Figma Designs

- Staking UX ([link](https://www.figma.com/design/1c0Y9jDJe6p0j82jydJDcs/Staking-UX?node-id=2979-19504&m=dev))

### **Before**

https://github.com/user-attachments/assets/bd1919d3-dcc6-4443-b08c-7944e183914c

### **After**

https://github.com/user-attachments/assets/924905b1-405a-49f9-aa9e-eb0cbeaf872e

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by amitabh94 — COMMENTED** [review]
Just a question and suggestion but otherwise good to go!

File: app/components/UI/Stake/sdk/stakeSdkProvider.tsx [hunk 1]
```diff
@@ -15,6 +17,9 @@ import { useSelector } from 'react-redux';
 import { selectChainId } from '../../../../selectors/networkController';
 import { hexToDecimal } from '../../../../util/conversions';
 
+const hexToChainId = (hex: `0x${string}`): ChainId =>
```
Comment (amitabh94) [inline app/components/UI/Stake/sdk/stakeSdkProvider.tsx:20]: You can also use getDecimalChainId() util function to make this easy

File: app/components/UI/Stake/__mocks__/mockData.ts [hunk 2]
```diff
@@ -86,3 +93,24 @@ export const MOCK_REWARD_DATA = {
...
+  encodeDepositTransactionData: jest.fn(),
+  encodeEnterExitQueueTransactionData: jest.fn(),
+  encodeMulticallTransactionData: jest.fn(),
+  estimateClaimExitedAssetsGas: jest.fn(),
+  estimateDepositGas: jest.fn(),
+  estimateEnterExitQueueGas: jest.fn(),
+  estimateMulticallGas: jest.fn(),
+};
+
+export const MOCK_POOL_STAKING_SDK: Stake = {
```
[the remark this answers is not in the document]
Comment (Matt561) [reply app/components/UI/Stake/__mocks__/mockData.ts:112]: Sure!

**Review by amitabh94 — APPROVED** [review]
LGTM!

**Pull request merged**